src/stdlib_sparse_conversion.fypp

```diff
@@ -0,0 +1,426 @@
+            do j = adr1, adr2
+                ELL%index(i,j-adr1+1) = CSR%col(j)
+                ELL%data(i,j-adr1+1)  = CSR%data(j)
+            end do
+        end do
+    end subroutine
+
+    #:endfor
+
+    #:for k1, t1, s1 in (KINDS_TYPES)
+    module subroutine csr2sellc_${s1}$(CSR,SELLC,chunk)
+        !> csr2sellc: This function enables transfering data from a CSR matrix to a SELL-C matrix
+        !> This algorithm was gracefully provided by Ivan Privec and adapted by Jose Alves
+        type(CSR_${s1}$), intent(in)      :: CSR
+        type(SELLC_${s1}$), intent(inout) :: SELLC
+        integer, intent(in), optional :: chunk
+        ${t1}$, parameter :: zero = zero_${s1}$
+        integer :: i, j, num_chunks
+
+        if(present(chunk)) SELLC%chunk_size = chunk
+
+        SELLC%nrows = CSR%nrows; SELLC%ncols = CSR%ncols
+        SELLC%storage = CSR%storage
+        associate( nrows=>SELLC%nrows, ncols=>SELLC%ncols, nnz=>SELLC%nnz, &
+        &         chunk_size=>SELLC%chunk_size     )
+        !-------------------------------------------
+        ! csr rowptr to SELL-C chunked rowptr
+        num_chunks = (nrows + chunk_size - 1)/chunk_size
+        allocate( SELLC%rowptr(num_chunks+1) )
+        block
+            integer :: cidx, rownnz, chunknnz
+            SELLC%rowptr(1) = 1
+            cidx = 1
+            do i = 1, nrows, chunk_size
+                chunknnz = 0
+                ! Iterate over rows in a given chunk
+                do j = i, min(i+chunk_size-1,nrows)
+                    rownnz = CSR%rowptr(j+1) - CSR%rowptr(j)
+                    chunknnz = max(chunknnz,rownnz)
+                end do
+                SELLC%rowptr(cidx+1) = SELLC%rowptr(cidx) + chunknnz
+                cidx = cidx + 1
+            end do
+            nnz = SELLC%rowptr(num_chunks+1) - 1
+        end block
+        !-------------------------------------------
+        ! copy values and colum index
+        allocate(SELLC%col(chunk_size,nnz), source = -1)
+        allocate(SELLC%data(chunk_size,nnz), source = zero )
+        block
+            integer :: lb, ri, iaa, iab, rownnz
+            do i = 1, num_chunks
+
+                lb = SELLC%rowptr(i)
+
+                ! Loop over rows of a chunk
+                do j = (i-1)*chunk_size + 1, min(i*chunk_size,nrows)
+    
+                    ri = j - (i - 1)*chunk_size
+                    
+                    rownnz = CSR%rowptr(j+1) - CSR%rowptr(j) - 1
+                    iaa    = CSR%rowptr(j)
+                    iab    = CSR%rowptr(j+1) - 1
+                    
+                    SELLC%col(ri,lb:lb+rownnz)  = CSR%col(iaa:iab)
+                    SELLC%data(ri,lb:lb+rownnz) = CSR%data(iaa:iab)
+                
+                end do
+            end do
+         end block
+        end associate
+    end subroutine
+
+    #:endfor
+
+    #:for k1, t1, s1 in (KINDS_TYPES)
+    recursive subroutine quicksort_i_${s1}$(a, b, first, last)
```

Comment:
would it be an idea to add such a procedure in `stdlib_sorting`, based on `sort_index`? `sort_index` is clearly limited for such simpler cases, and making the argument `index` as `intent(inout)` would already facilitate such cases.

Reply:
@jvdp1 I'm not sure if I got it right, you mean to say to add a quick sort within sort_index? Or to add it as a different method within the sort module?... I took a look at sort index but my first impression is that changing the index array to `inout` won't be enough. The internals seems to be thought for handling an array to sort and the work index arrays. Here, we need to sort one array and have a secondary array be sorted together with the first, simultaneously. From what I read in the specs, non of the current sorting interfaces enable this directly, but I just had quick glance so I might be missing something. I wouldn't mind trying to generalize the quicksort, or if you help me out figuring out how to adapt any other of the sorting procedures to enable sorting two arrays simultaneously (avoiding extra allocations), I could give it a try.

Reply:
I was thinking to extend `sort_index` such that a provided vector (other than just `integer(int64)`) is sorted in the same way as the sorted vector (if I am right, it is what you need). The kind of the array `index` is already generated by `fypp`. So I think that it should be possible (but I may be wrong). I will have a look.

Reply:
@jalvesz Could you maye test the `sort_index` in [this branch](https://github.com/jvdp1/stdlib/tree/extend_sort_index)? I did some dirty modifications such that the array `index` van be of any `integer` or `real` type. If it works, we will need to discuss names, and approaches, before getting a nice implementation.
Here is an example I tested in my branch:
```fortran
program example_sort_index_real
  use stdlib_sorting, only: sort_index
  implicit none
  integer, allocatable :: array(:)
  real, allocatable :: index(:)

  array = [5, 4, 3, 1, 10, 4, 9]
  index = real(array)

  call sort_index(array, index)

  print *, array   !print [1, 3, 4, 4, 5, 9, 10]
  print *, index   !print [1., 3., 4., 4., 5., 9., 10.]

end program example_sort_index_real
```

Reply:
Wow thanks! I'll test this out and let you know.

Reply:
@jvdp1 I tested your modifications to sort a COO matrix representing a 2millions nodes mesh, having 14millions of NNZ before sorting/removing duplicates:

Sorting index + data using quicksort (3 runs in release profile)
elapsed time (cpu_time) [s]: 1.4375559 / 1.4128339 / 1.1456219
Sorting index + data using sort_index:
elapsed time (cpu_time) [s]: 1.4756519 / 1.5304509 / 1.1852899

If I do not sort the data (only the index(1:2,1:nnz) array) then the stdlib `sort` is about 9% faster than the quicksort.

I would say these are more than satisfying results!

Now, I included support for complex sparse matrices and the sorting interfaces do not support them. So we should discuss what to do here to extend for complex data ( ? )

Reply:
> I would say these are more than satisfying results!

Nice. Thank you for testing it.

> Now, I included support for complex sparse matrices and the sorting interfaces do not support them. So we should discuss what to do here to extend for complex data ( ? )

It seems that complex data is not supported at all by the `stdlib` sorting procedures. It should be quite easy to add to all of them by extending the `fypp` variables to the complex  definitions.
Note that my changes broke the original `stdlib_index`. Now that I know that it works, I will have a closer look for a proper implementation.

Reply:
So I'm guessing this would deserve a nice PR on its own. Should we wait for that before this PR? or can we keep reviewing this knowing that this particular point can be improved down the road?

Reply:
You are right, it deserves its own PR. I suggest that you continue with your internal quicksort subroutine. We can review it later when the PR related to the sort procedures will be ready.

Reply:
See #849 for such an implementation